logos-codegen/src/mir.rs

```diff
@@ -110,9 +110,13 @@ impl TryFrom<Hir> for Mir {
                 }
 
                 let kind = repetition.kind;
+                let is_dot = *repetition.hir == Hir::dot(!repetition.hir.is_always_utf8());
```

Comment:
`Hir::dot` allocates. Can we put this into a `thread_local` or `once_cell` since it's always there?

Reply:
Yes, but as it wasn't already part of your dependencies, nor does it actually run during a program execution (only during compilation), I didn't feel like doing so. But I can easily make this change :)